src/linker/Linker.Steps/EmbeddedXmlStep.cs

```diff
@@ -37,76 +37,104 @@
 
 namespace Mono.Linker.Steps
 {
-
-	public class BlacklistStep : BaseStep
+	public class EmbeddedXmlStep : BasePerAssemblyStep
```

Comment:
Why does it need to be based on BasePerAssemblyStep. Couldn't be something simpler like `ProcessAssemblyEmbeddedXmls` with static ProcessXYZ methods?

Reply:
It could, but then we would need to pass the LinkContext and AssemblyDefinition through the static methods and all their helpers instead (same is true for the other per-assembly steps).